### ISSUE TITLE
proving that conneg is a breaking change

### ISSUE THREAD

**Issue by melvincarvalho** [w3c/WebID#58]
There are two Webid Specs, WebID 1.0 and WebID 2.0.  

WebID 1.0 mandates a serialization, say Turtle.

WebID 2.0 mandates conneg, with Turlte and JSON-LD

Alice has a WebID `alice.ttl` which has been her webid for 10 years.  She uses it with all her solid Apps.

One day Apps switch to WebID 2.0 and the newer JSON-LD becomes the preferred serialization

Alice's WebID is now broken.  Furthermore it's hosted on her own home page using Apache, and she has no ability to fix her webid.  This breaking change, makes all of Alice's work over the last 10 years potentially broken.

QED

**Comment by woutermont** [comment]
Really, @melvincarvalho? This is not an _issue_. @jacoscaz, I suggest moving this to Discussions, which we specifically opened for things like this.

I also think that we already had this discussion good and well in issue #3 regarding the serialisation formats. Before repeating ourselves here, I would thus suggest you (re)read the past months of that issue. In particular, I explained at length during multiple comments ([here](https://github.com/w3c/WebID/issues/3#issuecomment-1890801301), [here](https://github.com/w3c/WebID/issues/3#issuecomment-1890930810), [here](https://github.com/w3c/WebID/issues/3#issuecomment-1890938880), [here](https://github.com/w3c/WebID/issues/3#issuecomment-1890970585) and [here](https://github.com/w3c/WebID/issues/3#issuecomment-1891002060)), why switching to Turtle+JSON-LD using conneg would not break anything the WebID spec aims to support. In an effort to be constructive, I will try my best to summarize those points here as clear and succinct as I can (parts of the following text is copied from my aforementioned comments).

1. The vast majority of WebID users will be **non-technical people**, managing their WebID through the GUIs of **WebID providers**. This includes Alice and Bob, our main fictitious users.

2. All existing WebID providers (Inrupt, Use.ID, OpenLink, solidweb.me, redpencil.io, inrupt.net, solidcommunity.net ...) **already provide Turtle+JSON-LD through conneg**. (If I missed some, please tell me.) This is no wonder, because conneg is part of all major web frameworks, and no work at all to add to a server.

3. Given the above two points, <ins>**conneg does not break anything for the vast majority of WebID users**</ins>, and is thus perfectly in line with the main targets for a successful WebID ecosystem.

4. Now, if you really want to be nitpicky, there are indeed a few people who currently host a static WebID Document. They do so simply because they can: it has up till now been the simplest way to host a WebID. However, people that are skilled enough to host a WebID Document themselves are **easily skilled enough to set up conneg** over it. I've timed it: it takes about 5 minutes using a free CloudFlare account. I can guarantee you that the majority of these people will do so the instant we add conneg to the spec. Most importantly, however, these **these people are not the main target for the spec**. They are not the ones for which applications will be built and an ecosystem will grow; they are not the ones to which we must cater.

**Comment by melvincarvalho** [comment]
Thanks for this, replies inline.  I dont think discussions are universally liked, so perhaps lets keep it as an issue, for a bit:

> * The vast majority of WebID users will be **non-technical people**, managing their WebID through the GUIs of **WebID providers**. This includes Alice and Bob, our main fictitious users.

Doesnt really speak to breaking change

> 
> * All existing WebID providers (Inrupt, Use.ID, OpenLink, solidweb.me, redpencil.io, inrupt.net, solidcommunity.net ...) **already provide Turtle+JSON-LD through conneg**. (If I missed some, please tell me.) This is no wonder, because conneg is part of all major web frameworks, and no work at all to add to a server.

Not the example I gave of `alice.ttl` -- there's more than one such webid around.  So this is not a disproof

We dont even have a way to test that conneg works on the servers above.  Trust me, I've spent hours trying to design such a test with the "solid team".  They dont even understand conneg.  

> 
> * Given the above two points, **conneg does not break anything for the vast majority of WebID users**, and is thus perfectly in line with the main targets for a successful WebID ecosystem.

Again, the proof stands.

> 
> * Now, if you really want to be nitpicky, there are indeed a few people who currently host a static WebID Document. They do so simply because they can: it has up till now been the simplest way to host a WebID. However, people that are skilled enough to host a WebID Document themselves are **easily skilled enough to set up conneg** over it. I've timed it: it takes about 5 minutes using a free CloudFlare account. I can guarantee you that the majority of these people will do so the instant we add conneg to the spec. Most importantly, however, these **these people are not the main target for the spec**. They are not the ones for which applications will be built and an ecosystem will grow; they are not the ones to which we must cater.

I think you are under estimating the implementation burden here.  And how do you keep the different serializations in sync?

I think you are agreeing with me that it's a breaking change.  And you are trying to argue it's a minor change.  That is a subjective analysis, that I dont agree with.

But I thank you very much for taking the time to write this all out.  I do hope we can make progress.

**Comment by melvincarvalho** [comment]
How would this look for a compromise:

1. If you feel the changes are minor, then the new spec should be named with version `1.1` OR `2.0` but NOT `1.0`.  And the CG can later come to a consensus on which

2. Work on extension profiles can continue in the group, unblocking new work, and reducing dependencies on an all-or-nothing solution

Why dont we just agree is a breaking change, and make that very clear, through versioning, then unblock everyone in the group that wants to make progress.  Sound reasonable?

**Comment by woutermont** [comment]
I'm not agreeing at all. What I am trying to point out is that this is de facto a minor change (which is per definition not breaking), since none of the stakeholders at which this spec is aimed would be affected.

WebID is meant to provide an entry hook into the digital lives of people. To succeed in that, it will need the support of applications to become an ecosystem. That can only be achieved by being flexible on the server side.

The spec is NOT meant to serve the whims of some developers who want a static file and don't want to put in 5min of effort. If you want that, you can create another CG, though I doubt you will find many interested parties. But please stop setting your own desires above the aims of this group, who have finally come closer to consensus and are glad about it.

**Comment by woutermont** [comment]
> I think you are under estimating the implementation burden here. And how do you keep the different serializations in sync?

To shortly reply to this concern: as I said, I timed it, and it takes 5min. Keeping it in sync can be done with an automated script, which is about three lines of code. For example, as a GitHub workflow publishing to GitHub Pages, or using a tool like Soupault when running a static generator locally.

**Comment by woutermont** [comment]
> I think you are under estimating the implementation burden here. And how do you keep the different serializations in sync?

To shortly reply to this concern: as I said, I timed it, and it takes 5min. Keeping it in sync can be done with an automated script, which is about three lines of code. For example, as a GitHub workflow publishing to GitHub Pages, or using a tool like Soupault when running a static generator locally.

**Comment by woutermont** [comment]
@webr3, what you are saying makes no sense. Did you even take the time to read the statistics I put effort in to gather?

**All** current providers support conneg with both formats. Why? Because it is a no-brainer. So stop pulling the "conneg is a burden" card, please.

As I have clarified elsewhere, I have no ties to Turtle other than backward compatibility (which, by the way, is exactly what @melvincarvalho is making such a ruckus about). I fully agree with you that, in some future, it is possible and probable that the spec can drop Turtle support. But it will do so with a decent deprecation strategy, and only after the ecosystem has gradually moved to JSON-LD. Moreover, since the data format landscape has historically proven to be a volatile one, having conneg present will in the even farther future be a practical once more to move from JSON-LD to whatever the majority will fancy then. This is how we build a spec that can outlive such changes.

**Comment by melvincarvalho** [comment]
> So stop pulling the "conneg is a burden" card, please.

It's not only me:

See danbri, inventor of FOAF, that got the whole WebID movement going:

https://twitter.com/danbri/status/1080490927284736000

"the content negotiation aspect of linked data has been massively oversold"

I'm going to persist here, you will have to admit it's a breaking change in the case of `alice.ttl` or give me a reason why it is not.  After that, we can proceed.

**Comment by woutermont** [comment]
@webr

> Did you time conneging flat files, or did you implement something like a mongodb to turtle implementation with media type juggling that produces valid turtle in 5 mins?

Does it matter? The point is that those handful of people that want it can easily find ways to do it, and at least one of them is set up in a jiffy.

My test in particular was aimed at individuals hosting static files. So we're talking about pushing one format to GitHub Pages, having a workflow transpile to the other format, and setting up a Cloudflare rule that redirects to either of them depending on the Accept header.

For people setting up a database to host their (and presumably others) WebID data, adding conneg should even be less of a burden, relative to the time it already takes to set up the database in the first place.

**Comment by woutermont** [comment]
> > So stop pulling the "conneg is a burden" card, please
> 
> It is. You offering proof of existing rdf conneg compatible systems doing what they do as "easy", of course it is, that's what they do.

That is not my argument. What I show is that, without the spec asking or even suggesting to do so, ALL WebID providers decided to add conneg. So either all of them happen to have massive resources on their hands, or adding conneg really isn't that hard... Which one would it be?

> Whilst completely ignoring the requirement to turn non-rdf non-conneg systems in to doing all that, in order to adopt WebID. That is big, and it's very preclusive to adoption.
> 
> You cannot say that tiktok or Snapchat or FB quest Auth can just rdf-ize and conneg their systems "in 5 minutes" to adopt it.

I fully miss what your argument is here. The services you mention do not implement WebID at all. If they would want to do so, they would have to make an investment into RDF anyway. Adding conneg to that investment will be like adding one pea to a bag full of peas.

**Comment by woutermont** [comment]
> The point is being missed.
> 
> Conneg = super easy
> Conneg to valid RDF = burdensome.

Oh, please explain...

**Comment by melvincarvalho** [comment]
> at least one of them is set up in a jiffy

What is the URL?

**Comment by woutermont** [comment]
Using JSON-LD without RDF stack is a thing, but unless you have numbers I don't believe for a second it is widespread. It's not like non-RDF services just happen to have JSON-LD WebID data lying around, and we're forcing them to actually treat it as RDF. 

Moreover, even for players that want to enter the WebID ecosystem and up till then work with "non-RDF JSON-LD", adding the few lines of code to transpile valid JSON-LD to valid Turtle can hardly be called a burden.

You seem to have a grudge against Solid for some reason, but for all the years that WebID existed, they are the ones that actually put it on the map. As far as I know all WebID providers except OpenLink are primarily Solid businesses or organisations.

**Comment by woutermont** [comment]
Wait, what kind of documents/domains are we talking about? Numbers mean nothing by just throwing them out without reference. These surely cannot be WebIDs or WebID-aspiring services?

I also don't see where you want this to go? WebID has been an RDF standard from the start. It currently is produced as Turtle, and I agree that consuming them as JSON-LD would be a benefit for lots of applications. That is precisely why we are proposing conneg. You surely are not saying that applications preferring JSON-LD cannot add an accept header to their requests?

**Comment by woutermont** [comment]
To go into the only concrete example you gave: CMS's that can produce schema.org data (whatever that is).

Schema.org is (or at least pretends) to be an RDF vocabulary, so services producing such output produce RDF, willing or not (just like all JSON-LD is RDF). As I explained above, there is nothing preventing these systems to use one of the widely available RDF transpilers to output any other RDF format on request.

Moreover, I would actively dissuade anyone from using a CMS to host their WebID Document. They are absolutely not secure enough for such a central target in online authentication.

**Comment by woutermont** [comment]
Anyway, I'm going to wait for others to chip in. I find it deplorable, however, that you would put a hard-reached consensus, of which you yourself said it was a way forward, on the line again.

**Comment by woutermont** [comment]
@webr3, can I ask what the service is your client is providing? More particularly, I am curious as to why they would want to host WebIDs themselves. 

As to the scalability of content negotiated RDF, multiple Solid stakeholders seem to disagree with you (given that they are doing it, I assume they believe it is possible). Without clear insight into your setup, or a benchmark to prove otherwise, I therefore still find it hard to believe.

**Comment by jacoscaz** [comment]
_/chair hat off_

> If you feel the changes are minor, then the new spec should be named with version 1.1 OR 2.0 but NOT 1.0.

@melvincarvalho to clarify, would changing the title of the working draft https://w3c.github.io/WebID/spec/identity/index.html to _Web Identity and Discovery **1.1**_ cause you to withdraw your strong objection to conneg?

**Comment by kidehen** [comment]
> WebID is meant to provide an entry hook into the digital lives of people. To succeed in that, it will need the support of applications to become an ecosystem. That can only be achieved by being flexible on the server side.

Yes, and this can be achieved  while also relegating content-negotiation to the implementation details level i.e., putting it in the implementation example (or techniques) section as one approach. 

I agree with where you are headed, but content-negotiation doesn't need to be a part of the spec, since its just a technique afforded by HTTP re server implementation.

**Comment by melvincarvalho** [comment]
> _/chair hat on_
> 
> > If you feel the changes are minor, then the new spec should be named with version 1.1 OR 2.0 but NOT 1.0.
> 
> @melvincarvalho to clarify, would changing the title of the working draft https://w3c.github.io/WebID/spec/identity/index.html to _Web Identity and Discovery **1.1**_ cause you to withdraw your strong objection to conneg?

I appreciate the ongoing discussions and the commitment of everyone involved. I'd like to clarify my perspective regarding the recent propositions:

1. **Proposal Context**: Initially, the suggestion of simultaneously pursuing v2.0 or v1.1 alongside extension profiles seemed viable. However, my understanding evolved following the insights about the potential [deprecation of Turtle](https://github.com/w3c/WebID/issues/58#issuecomment-1925424984).

2. **Acknowledging Breaking Changes**: It's crucial for the group, including Wouter, to recognize that mandating content negotiation (conneg) constitutes a significant modification. This point, I believe, should have been the preliminary observation. I've illustrated this in our discussions, yet I feel the responses have somewhat downplayed its impact.

3. **Implications on Versioning**: Acknowledging the substantial nature of these changes naturally leads to the question of versioning. Given the scale, it seems appropriate to consider this a move towards v2.0. The discourse isn't merely about introducing JSON-LD but about a gradual phase-out of Turtle, marking a substantial shift.

4. **Consensus and Versioning Integrity**: I'm concerned that the principle of lazy consensus might be misapplied to introduce significant alterations without the due process of version incrementation. For a spec that has reached a level of stability and fostered a wide ecosystem, any such major change warrants a more deliberate and inclusive consensus-building process.

5. **Process and Controversy**: The collective effort over the past six months was grounded in achieving a shared understanding of WebID, independent of serialization, which is foundational. The notion of extension profiles was an extension of this consensus, aiming to cater to diverse requirements. The shift in focus towards altering a stable spec, however, diverges from this path and delves into areas of long-standing debate. It's crucial to tread these waters with consensus and respect for the broader implications.

6. **Path Forward**: If we are to consider any significant changes or deprecation of Turtle, it necessitates broad consensus, including dialogues within the Solid WG. Such changes should be transparently proposed, with explicit versioning considerations. A living standard cannot be retrofitted with fundamental changes without a clear, collective agreement.

In summary, I urge us to:
- Validate that mandating conneg and possibly deprecating Turtle indeed marks a breaking change.
- Ensure any breaking change is accompanied by a major version increment.
- Provide viable alternatives, such as extension profiles, to maintain a functional spec.
- Engage the Solid WG and, ideally, seek insights from timbl regarding the deprecation of Turtle.

@jacoscaz, your leadership is invaluable, and this isn't a reflection on your chairmanship. Steering the direction of our work requires sensitivity to the foundational elements of the spec, and significant changes demand a proportionate consensus-building effort.  Changing things that have been in use for 10 years is going to be 10x more difficult than making progress on a new fresh, *extensible*, version of WebID for 2024, which gives everyone what they want.

**Comment by namedgraph** [comment]
> I think you are under estimating the implementation burden here. And how do you keep the different serializations in sync?

@melvincarvalho respectfully, this quote alone proves that you don't understand how RDF-aware applications are built.
No serious Linked Data application would store RDF as _documents_, with a copy per media type. Which would lead to issues like them going out of sync.

RDF is being stored as _triples_ in a triplestore, an abstract graph model that has no inherent syntax. Only upon HTTP response the right media type is chosen using conneg and used to serialize the response body. Conneg is automatically handled by the HTTP framework such as Jersey. Switching serializations is as simply as passing different media types to Jena's I/O. From the client perspective there are still _documents_, but they are generated on the fly (and can be cached of course).

**Comment by namedgraph** [comment]
Now lets try to illustrate this issue with some examples:

## WebID 1.0 request/response

```
GET alice.ttl
Accept: application/ld+json, text/turtle, */*;q=0.8

200 OK
Content-Type: text/turtle
```

## WebID 2.0 request/response

```
GET alice.ttl
Accept: application/ld+json, text/turtle, */*;q=0.8

200 OK
Content-Type: text/turtle
```

It looks to me that the same headers would work with both WebID 1.0 and 2.0?

**Comment by melvincarvalho** [comment]
> respectfully, this quote alone proves that you don't understand how RDF-aware applications are built.

@namedgraph this is not respectful.  I have coded many RDF-aware applications.  It's an ad hominem attack and unprofessional.  Please retract.

**Comment by namedgraph** [comment]
OK, comment updated

**Comment by melvincarvalho** [comment]
> no serious Linked Data application would store RDF as _documents_

Sure they would.  Linked Data is just another name for the Semantic Web.  It's about having machine and human readable documents on the web, including with hyperlinks.

The style you may be your personal preference, but you need not project that preference onto others.

Let me ask you, how many daily active users do your Linked Data applications have (or monthly if daily=0).  What do they use them for?  Why do you consider your view to be a universal one, who are you representing?  

Id suggest making your tone more subjective, since there are obviously a range or publishing techniques in RDF.

**Comment by melvincarvalho** [comment]
> Now lets try to illustrate this issue with some examples:
> ## WebID 1.0 request/response
> 
> ```
> GET alice.ttl
> Accept: application/ld+json, text/turtle, */*;q=0.8
> 
> 200 OK
> Content-Type: text/turtle
> ```
> 
> ## WebID 2.0 request/response
> 
> ```
> GET alice.ttl
> Accept: application/ld+json, text/turtle, */*;q=0.8
> 
> 200 OK
> Content-Type: text/turtle
> ```
> 
> It looks to me that the same headers would work with both WebID 1.0 and 2.0?

If the application is expecting JSON-LD then the turtle only response will fail.  That's why JSON-LD=MUST is a breaking change.

**Comment by woutermont** [comment]
> If the application is expecting JSON-LD then the turtle only response will fail. That's why JSON-LD=MUST is a breaking change.

Exaclty. This is why we need conneg.

**Comment by melvincarvalho** [comment]
> > If the application is expecting JSON-LD then the turtle only response will fail. That's why JSON-LD=MUST is a breaking change.
> 
> Exactly. This is why we need conneg.

I believe conneg is implied here.  But this issue is simply proving, in a pseudo-mathematical, sense that mandating conneg (implicitly or explicit) via dual MUSTs for Turlte AND JSON-LD is a breaking change.  I think it's something we could agree on.

**Comment by namedgraph** [comment]
> > It looks to me that the same headers would work with both WebID 1.0 and 2.0?
> 
> If the application is expecting JSON-LD then the turtle only response will fail. That's why JSON-LD=MUST is a breaking change.

Isn't WebID 2.0 mandating both JSON-LD _and_ Turtle? In which case the client should be able to handle both.

**Comment by namedgraph** [comment]
> > no serious Linked Data application would store RDF as _documents_
> 
> Sure they would. Linked Data is just another name for the Semantic Web. It's about having machine and human readable documents on the web, including with hyperlinks.
> 
> The style you may be your personal preference, but you need not project that preference onto others.

Melvin, this is not even RDF or Semantic Web specific. There was a time 20-30 years ago where websites consisted of static HTML documents. Nowadays the absolute majority of webpages and API responses are generated from databases on the fly, as I'm sure you would agree.
The same applies to Linked Data, where the database is the triplestore. RDF has an additional advantage of offering a number of syntaxes for the same graph content, and conneg allows clients to choose them.
 
> Let me ask you, how many daily active users do your Linked Data applications have (or monthly if daily=0). What do they use them for? Why do you consider your view to be a universal one, who are you representing?

It's hard to tell what the users are using our software for, but it does have 450+ GH stars at the moment: https://github.com/AtomGraph/LinkedDataHub
It has supported conneg for 10+ years since the [project that was an early prototype of LinkedDataHub](https://2022-eu.semantics.cc/eldc-awards-given).

> 
> Id suggest making your tone more subjective, since there are obviously a range or publishing techniques in RDF.

**Comment by melvincarvalho** [comment]
@namedgraph thanks for the additional context.  That is a reasonable deployment.  Yet I dont think you speak for everyone.  It's absolutely legitimate to have documents on the web, that's how the web started, and how millions of people use the web.

Do you agree that the addition of conneg=MUST is a breaking change on the server side?

Clients may only support 1 of 2 serializations.  And, as wouter says, turtle itself my be deprecated.

**Comment by melvincarvalho** [comment]
> > > It looks to me that the same headers would work with both WebID 1.0 and 2.0?
> > 
> > 
> > If the application is expecting JSON-LD then the turtle only response will fail. That's why JSON-LD=MUST is a breaking change.
> 
> Isn't WebID 2.0 mandating both JSON-LD _and_ Turtle? In which case the client should be able to handle both.

Client only needs to be able to handle 1 if the server handles both.  There's the point of breakage.

**Comment by namedgraph** [comment]
Melvin I think I know what you're hinting at but I still don't agree.

You are suggesting a situation where a client only supports JSON-LD, correct?
Well first of all it shouldn't request Turtle if it cannot handle it. Lets model that:

```
GET alice.ttl
Accept: application/ld+json

406 Not Acceptable
```
Like this? Yes JSON-LD-only client would fail to load the Turtle data in this case. But that would also be the case with WebID 1.0, so I don't understand what you think changes here?

**Comment by melvincarvalho** [comment]
> Melvin I think I know what you're hinting at but I still don't agree.
> 
> You are suggesting a situation where a client only supports JSON-LD, correct? Well first of all it shouldn't request Turtle if it cannot handle it. Lets model that:
> 
> ```
> GET alice.ttl
> Accept: application/ld+json
> 
> 406 Not Acceptable
> ```
> 
> Like this? Yes JSON-LD-only client would fail to load the Turtle data in this case. But that would also be the case with WebID 1.0, so I don't understand what you think changes here?

Yes, exactly.

In WebID 1.0 everything works because you have a common ground, namely, turlte.

If there is a change to BOTH turtle and JSON-LD being MUST (e.g. via conneg) then the client can be either turlte or json.

That means that webid 1.0 will not work with the new webid 2.0.  Hence its a breaking change.

**Comment by namedgraph** [comment]
I would argue (as I've done before) that the issue here is that 1.0 specified _any_ serialization at all, in this case Turtle. Because we're talking about an orthogonal HTTP/conneg issue that is not specific to WebID.

TL;DR: clients and servers might negotiate and might not find an acceptable serialization, and that is simply a common HTTP error and not a WebID compliance issue

In other words, conneg is the _solution_ here, not the problem. The server has a set of media types it can read/write, as does the client. Conneg allows them to find an intersection between those sets, or if there is none, leads to an error. And that works perfectly transparently -- until someone puts some text in a specification like WebID that some specific serialization _MUST_ be supported, which breaks the orthogonality.

**Comment by melvincarvalho** [comment]
> _/chair hat on_
> 
> > If you feel the changes are minor, then the new spec should be named with version 1.1 OR 2.0 but NOT 1.0.
> 
> @melvincarvalho to clarify, would changing the title of the working draft https://w3c.github.io/WebID/spec/identity/index.html to _Web Identity and Discovery **1.1**_ cause you to withdraw your strong objection to conneg?

Concretely, there are multiple aspects

1. **Advocacy for Status Quo**: as expressed in the call for a complete [freeze](https://lists.w3.org/Archives/Public/public-webid/2024Jan/0030.html). This stance acknowledges the longstanding impasse within the group. However, I also appreciate that the formation of a Solid WG could introduce pivotal change, potentially dissolving the freeze.

2. **Versioning for New Initiatives**: I suggest that any new endeavors stemming from the WebID ED 2014 adopt a fresh versioning approach. This clear demarcation would properly represent the involvement of new authors, editors, and methodologies. For substantial revisions, version 2.0 seems fitting, whereas 1.x might suit more incremental updates. It's worth noting that introducing mandatory content negotiation (conneg) is a significant alteration and should be treated as such in versioning.  This issue is about proving that the change clearly is breaking.

3. **Personal Stance on Conneg**: On a personal note, the push towards mandatory conneg is a significant shift that I find challenging to align with, and could not live with, if it was the only choice. That said, exploring the development of a JSON-LD extension profile. Detailed guidance on its implementation, naming conventions, and structure would offer a constructive pathway. This approach would also allow those inclined towards a conneg path to proceed independently, ensuring diverse yet harmonious development within the community.

I hope that is clear enough.  It can be revisited this when the Solid WG starts up.  Mandating conneg is by far the most controversial change that has been suggested to the group in many years.   Replacing RDFa with JSON-LD and structured data islands would have been far easier.  If you want to make a new draft, with new authors, and a new set of defaults, bump the version to 2.0, and see if it stands on its own merits.

**Comment by melvincarvalho** [comment]
> I would argue (as I've done before) that the issue here is that 1.0 specified _any_ serialization at all, in this case Turtle. Because we're talking about an orthogonal HTTP/conneg issue that is not specific to WebID.

So, Turtle was specifically chosen as a MUST in WebID 1.0 because it guarantees interop.

The vast majority of the web does not use conneg.  And WebID is designed as a "Web" technology.

Yuo need a common ground so that everything works together.  

Turtle was chosen, maybe that was a mistake in retrospect, but it was the decision taken.  A change to that would be breaking.

**Comment by namedgraph** [comment]
On the flip side, WebID document resolution depends on HTTP (as does the rest of the web) and HTTP supports conneg by default. So yes, it was a mistake to attempt to override the default HTTP behaviour because it was clearly out of scope for the WebID spec.
That has now put us in a bind trying to satisfy conflicting requirements: Turtle is `MUST` due to backwards compatibility with 1.0, and JSON-LD is `MUST` due to new requirements. I don't see how these can be reconciled in the light of your example.

**Comment by namedgraph** [comment]
> Turtle was chosen, maybe that was a mistake in retrospect, but it was the decision taken. A change to that would be breaking.

If we would relax the serialization requirements by removing Turtle, i.e. removing them altogether, how would that break the backwards compatibility?

**Comment by melvincarvalho** [comment]
> > Turtle was chosen, maybe that was a mistake in retrospect, but it was the decision taken. A change to that would be breaking.
> 
> If we would relax the serialization requirements by removing Turtle, i.e. removing them altogether, how would that break the backwards compatibility?

Because there's no common ground for libraries to ensure everything works for all users.  That's why ONE serialization was selected.  Turtle just seemed a good choice at the time.

**Comment by namedgraph** [comment]
That was then, but I'm
talking about making a change now.

No other spec in the RDF stack such as SPARQL or SHACL specifies any mandatory RDF serializations, because it is understood it's a completely orthogonal concern.

**Comment by melvincarvalho** [comment]
> That was then, but I'm talking about making a change now.
> 
> No other spec in the RDF stack such as SPARQL or SHACL specifies any mandatory RDF serializations, because it is understood it's a completely orthogonal concern.

I believe solid does.

I've sent a message to the Solid CG to see if anyone has strong views on Turtle vs JSON-LD:

https://lists.w3.org/Archives/Public/public-solid/2024Feb/0001.html

Let's see what they say.

**Comment by melvincarvalho** [comment]
> @melvincarvalho to clarify, would changing the title of the working draft https://w3c.github.io/WebID/spec/identity/index.html to _Web Identity and Discovery **1.1**_ cause you to withdraw your strong objection to conneg?

Perhaps its helpful to give some background.  WebID was in development for 5+ years before it got to 1.0 in 2014.  Longer if you consider foaf and foaf-protocols before it.

The reason it was called 1.0 rather than 0.x was because it was considered finished, stable, and ready to use by the authors.  That proved to be the case as we spent many years building Solid on top of it.  The only reason it didnt become a REC was because of an arbitrary decision of LDP not to include identity in the first version.  There was then an idea it could become a REC in the Solid WG.

That's why it's 1.0 and not 0.10 etc.  Note Solid is still in the 0.x phase which indicates that 0.x+1 can change bits.  1.0 was an indicator of stability and completion and it has been so for 10 years.

Any new work with new authors/editors/methodology should respect that by starting at 1.1 under all scenarios.  It's just far too confusing to do anything else.  Minor changes could be put into a 1.1 spec and subsequently published.  The main thing that I see is that JSON-LD could be put on the same footing as RDFa as a MAY.  In fact RDFa should be replaced with JSON-LD IMHO.   That I can see being a non-obtrusive 1.1 release.

Any major breaking changes should be 2.x, and conneg being mandated falls under that category.

The only scenario I can see where something different could apply would be if everyone, including the Solid WG had a unanimous agreement to redefine the concept in order to make it a W3C REC.  But we dont have word from the Solid WG yet, so that seems premature.

Let's do proper versioning based on breaking changes, and to show respect to the authors that came before who felt, understandably, that they had made something finished and ready to use.  I hope that adds some extra clarity.

**Comment by melvincarvalho** [comment]
> If we would relax the serialization requirements by removing Turtle, i.e. removing them altogether, how would that break the backwards compatibility?

Timbl was very insistent on having at least one MUST.  Basically it means that all clients and servers can ensure a consistent UX for the eco system.

Let's say you dropped turlte, and then one group wanted RDF/XML another n3.  End users would not know what was happening.

**Comment by jacoscaz** [comment]
_/chair hat on_

There are a few things that I would kindly ask everyone to stop and think about before continuing this conversation.

Quoting from [this comment](https://github.com/w3c/WebID/issues/58#issuecomment-1925411858):

> All existing WebID providers (Inrupt, Use.ID, OpenLink, solidweb.me, redpencil.io, inrupt.net, solidcommunity.net ...) already provide Turtle+JSON-LD through conneg.

Quoting from [this comment](https://github.com/w3c/WebID/issues/3#issuecomment-1890900892):

> I've search through the 6p of NPM packages that use the term "WebID" in their description, and looked into their actual code. Here are the numbers:
> 
> - Only 7 packages actually looked up a WebID Document.
> - Only 2 packages them are popular (>10 downloads/week):
>     - @inrupt/solid-client-js (9,433): requests & parses Turtle (though caller code can pass in another parser).
>     - @solid/access-token-verifier(395): requests & parses Turtle
> - 71.4% (5/7) only request & parse Turtle
> - Only 2 hardly used packages can parse something else:
>     - webid-pubkey-credentials-module(3): requests unspecified & parses any RDF format
>     - node-webid (0): requests Turtle or JSON-LD & parses any RDF format
> - webid-provider-tests (of the Solid CGs official test suite): requests & parses Turtle
>
> Again, in as far as these numbers tell us anything, they clearly indicate the predominant presence of Turtle as expected WebID format, not only in absolute numbers, but especially through the strong reliance on a limited set of libraries that use Turtle.

Unless anyone can refute the above, the only practical breaking change that would be introduced by a MUST on Turtle and JSON-LD for publishers would be to make statically hosted WebID Profile Documents in the form of Turtle `.ttl` files non-compliant.

This is, indeed, a breaking change, albeit a minor one if we weigh it against the number of occurrences of this particular hosting scenario. Nonetheless, it deserves to be called for what it is - breaking. If we were dealing with a WG aiming to push forward with the next version of a published spec, this would amply justify a version increase.

However, as the [2014 ED](https://w3c.github.io/WebID/spec/drafts/ED-webid-20140305/identity/) itself reminds us, **there is no such a thing as WebID 1.0**:

> This document is produced from work by the [W3C WebID Community Group](http://www.w3.org/community/webid/). This is an internal draft document and may not even end up being officially published. It may also be updated, replaced or obsoleted by other documents at any time. It is inappropriate to cite this document as other than work in progress. The source code for this document is available at the following URI: https://dvcs.w3.org/hg/WebID

In fact, any WG that adopted our drafts would be in charge of producing exactly that, the first published, official version of the WebID spec: _Web Identity and Discovery (WebID) **1.0**_ . Adding a version number other than 1.0 to our drafts makes absolutely no sense, if only for the fact that the first official version of the spec would see that number _decrease_ back to 1.0.

Nonetheless, given that we _are_ dealing with a breaking change, we do need to provide some way to easily discriminate the working draft from the 2014 ED. Luckily enough, this is how the 2014 ED looks, both in this repo and [in the old incubator pages](https://www.w3.org/2005/Incubator/webid/spec/identity/):

<img width="1440" alt="Screenshot 2024-02-04 at 16 54 39" src="https://github.com/w3c/WebID/assets/3322119/88ce8824-56e1-45ca-9cdf-4669099ea1a7">

Though it has been around for a long time, any adopter of the WebID 2014 ED who has actually taken the time to even simply glance at the spec from a distance _is already aware of its nature as an ED_ because the ED itself makes that crystal clear.

**There is no reason to hold back on a breaking change, if there is consensus behind it, as long as we produce a document that can be easily told apart from the 2014 ED. As the screenshot above demonstrates, that's more than easily achieved by simply changing the date.**

This is how I'm going to move forward, as chair. If any of you were to find it unpalatable, I invite you to a) refute any of the above in a clear, demonstrable, replicable, grounded manner and b) ask me to resign. I serve the group and, if the group deems it so, I'm happy to step down and let someone else carry things forward.

**Comment by melvincarvalho** [comment]
> This is, indeed, a breaking change,

Thank you very much for confirming this.  That was the purpose of the issue.

Also thank you for considering all the nuanced viewpoints and reaching that conclusion.  IMHO an example of good chairing.

I am saying very clearly that I do not want breaking changes to the spec that we have, no matter how slight they care considered by some, there is clearly a spectrum of views on how big a change it is.  I personally feel it's the biggest breaking suggestion we've had in years.

> There is no reason to hold back on a breaking change

I disagree on this completely.  You do NOT have consensus.

> This is how I'm going to move forward, as chair.

You cannot railroad breaking changes through to a mature, in use, 10 year stable spec, which has a eco systems built up around it, without full consensus.

I simply ask you to focus on items with consensus.

**Comment by melvincarvalho** [comment]
> As the screenshot above demonstrates, that's more than easily achieved by simply changing the date.

You honestly lost me completely wtth this.

![image](https://github.com/w3c/WebID/assets/65864/5042a618-38fc-40df-99c5-ee7caa58694b)

It clearly says WebID 1.0 on your own diagram.  It was considered finished, complete and ready for production by the authors.  Changing the date to shoe-horn in breaking changes is a non-starter.

**Comment by melvincarvalho** [comment]
This issue can now be closed, I think we've all expressed ourselves.

We can move forward having proved, and with a group consensus that introducing JSON-LD=MUST and/or conneg=MUST is a breaking change.

It goes without saying that breaking changes are a high, or perhaps the highest, level of controversial changes a group can have.

How we move forward I think is an orthogonal conversation.

So I suggest closing in this in a bit, unless someone wants to try and argue that it's not a breaking change.  Else we can move forward with a shared understanding which I think will enable progress.

**Comment by jacoscaz** [comment]
> You cannot railroad breaking changes through to a mature, in use, 10 year stable spec, which has a eco systems built up around it, without full consensus.

@melvincarvalho at this point I'm not going to spend any further time on arguing about whether I'm railroading anything or not. Fortunately, everything we do is on record, for everyone to see. You have the right to draw your own conclusions, of course, and I respect that. We obviously disagree on many things, which is also absolutely fine, but I do not have it in myself to continue this manner of interaction as I'm aware that I would become more adversarial than anyone ought to be in their professional (and, IMHO, personal) conduct.

I'm going to move forward following the principles that I've laid out in https://github.com/w3c/WebID/issues/58#issuecomment-1925809369 unless someone refutes them in a clear, demonstrable, replicable, grounded manner and/or the group asks me to step down.

**Comment by melvincarvalho** [comment]
> @melvincarvalho at this point I'm not going to spend any further time on arguing about whether I'm railroading anything or not. Fortunately, everything we do is on record, for everyone to see. You have the right to draw your own conclusions, of course, and I respect that. We obviously disagree on many things, which is also absolutely fine, but I do not have it in myself to continue this manner of interaction as I'm aware that I would become more adversarial than anyone ought to be in their professional (and, IMHO, personal) conduct.
> 
> I'm going to move forward following the principles that I've laid out in [#58 (comment)](https://github.com/w3c/WebID/issues/58#issuecomment-1925809369) unless someone refutes them in a clear, demonstrable, replicable, grounded manner and/or the group asks me to step down.

- You do not have consensus for that.  I would appreciate if you would at least acknowledge that.

Counter proposal.  Why not do things that went through the hard multi-month, multi year process of gaining consensus such as the superspec, and extension profiles?

- You do not have consensus for breaking changes.  I would appreciate if you would at least acknowledge that.

Counter proposal.  Why not do things that are non-breaking, or completely new work?

- You do not have consensus to reuse the existing 1.0 version number for your newly edited work.  I would appreciate if you would at least acknowledge that.

Counter proposal: bump the version number so that ALL can see the difference.  Why cant you do this?  You didnt write WebID 1.0, what claim do you have to it?

- You have yet to receive consensus from future Solid WG for your changes.  I would appreciate if you would at least acknowledge that.

Counter proposal:  why not be patient and wait for Solid WG to start up, and engage with them, ask them what they want.  See if they want WebID to be a REC?  Why does it have to be so rushed through (you perhaps dont like my use of the word railroading, so i could withdraw it, and say "rushed", is that better?)

How you proceed is up to you.  But I will say that you have a close to perfect track record so far.  I can only ask you take consensus into account with your decisions, and Im optimistic that you will.

I would urge you to focus on the items where you do have consensus.  IMHO you've done a fantastic job so far.  You've just pivoted from the tricky task of making WebID modular and useful, to the impossible task of trying to gain the most difficult consensus that has not been achieved in years, and proposing the biggest change.

I'd entourage you to take the low hanging fruit.

**Comment by kidehen** [comment]
> Sure they would. Linked Data is just another name for the Semantic Web. It's about having machine and human readable documents on the web, including with hyperlinks.

Linked Data isn't another name for the Semantic Web. 

Linked Data is simple a principled approach to structured data representation that CAN manifest a Semantic Web.  The rules are simple:

1. Name anything of interest using an HTTP URI.
2. Describe said item of interest using RDF sentences/statements -- where subjects and predicates are denoted by an HTTP URI, while objects are denoted by an HTTP URI or literals (which may be typed or untyped).

A Semantic Web is an entity relationship graph comprising machine-readable (or computable) entity relationship type semantics -- informed by an Ontology. 

Thus, you can create a basic "Web of Data" using Linked Data principles without links that denote terms in an Ontology (the key to machine-readable entity relationship type semantics).

**Comment by kidehen** [comment]
> * provide _examples_ in multiple mediatypes (extension profiles).

Examples can be provided in the specs too, without the need for profiles. 
The issue right now is that the old spec conflates too many things, for all the wrong reasons.

**Comment by melvincarvalho** [comment]
Ive suggested this as a possible direction to take.

https://lists.w3.org/Archives/Public/public-webid/2024Feb/0002.html

**Comment by melvincarvalho** [comment]
> This is how I'm going to move forward, as chair. If any of you were to find it unpalatable, I invite you to a) refute any of the above in a clear, demonstrable, replicable, grounded manner and b) ask me to resign. I serve the group and, if the group deems it so, I'm happy to step down and let someone else carry things forward.

Your leadership as chair has been nothing short of remarkable, propelling us to unprecedented achievements this quarter. Your ability to discern and address legitimate concerns has been a key factor in our success. 

However, I would much rather suggest a pause, than anything more drastic.   In order to allow the soon-to-form Solid WG to shape, and they can inform us with vital new information. Taking a well-deserved break might be far preferable to resigning or prematurely tackling highly controversial items.  The problems then start to fix themselves in Q2 of 2024 when there will be clarity, where before there was none.

The progress we've made under your guidance is undeniable, and what is needed is a serialization and implementation neutral way to define and specify WebID.  Conversely, without that shared understanding, we stand almost no chance of creating a consensus based specification.  So the only thing needed here is to do things in the right order, and if in doubt, to slow down.  I appreciate that is easier said than done, and that you take pride in your work.  But let's build on success in a low-risk way, rather than taking the (unnecessary) high risk path to premature failure.

**Comment by TallTed** [comment]
> slowing down or pausing

Indeed. 65 comments on one new issue over a weekend. Plus at least two significant email threads. And other (long!) issue discussions besides.

It's great that some people are able to devote all their waking time to such. I am not one of those people.

Slowing down, giving time for readers to consider and respond to comments/posts, would not only be a good idea, I think it a necessity. Else, this may well turn into a 5-ish person round-robin, the output of which may not find general consensus among those in the CG (or eventual WG), never mind eventual team review for transition to REC.

**Comment by melvincarvalho** [comment]
Discussions are fine, as they refine understanding.  For example it was unclear before this issue, that proposed changes to add JSON-LD and conneg as a MUST were a breaking change.

This issue has established that the proposed changes are indeed a breaking change

What should be slowed or paused is breaking changes being applied to WebID.  I have advocated for a freeze for some time, until we can, as a group, come up with a [serialization and implementation neutral way to define and specify WebID](https://lists.w3.org/Archives/Public/public-webid/2024Feb/0002.html).  Something that we as a group should have done years ago.

If that can be achieved in Q1 of 2023, or even Q2 (there is no rush).  It would be a valuable documentation that lives on its own merits.  Then, and only then, is it logical to discuss serialization preferences.

**Comment by melvincarvalho** [comment]
> It's great that some people are able to devote all their waking time to such.

@TallTed please, be respectful.  Your tone is inappropriate.  Passive aggressive micro-aggressions add nothing constructive to the conversation.

**Comment by melvincarvalho** [comment]
> Really, @melvincarvalho? This is not an _issue_. @jacoscaz, I suggest moving this to Discussions, which we specifically opened for things like this.

I dont mind at this point if this is converted to a discussion, or closed, having established the main premise.

**Comment by jacoscaz** [comment]
_/chair hat on_

> Else, this may well turn into a 5-ish person round-robin, the output of which may not find general consensus among those in the CG (or eventual WG), never mind eventual team review for transition to REC.

@TallTed and all: we do have a [process](https://github.com/w3c/WebID/blob/main/CONTRIBUTING.md#processing-prs) in place that guarantees review times of up to 4 weeks for breaking changes and I remain steadfast in my quest for unanimity first and, if that cannot be achieved, as large a consensus as we can muster. While conversation can happen in bursts and evolve quickly, this doesn't automatically extend to how changes are made.

> Else, this may well turn into a 5-ish person round-robin, the output of which may not find general consensus among those in the CG (or eventual WG), never mind eventual team review for transition to REC.

There's really no risk of this happening as long as we stick to the process we have in place. While I do share a certain frustration with how parts of the conversation are being brought forward, at the present moment there's no risk of anything being decided against general consensus.

**Comment by melvincarvalho** [comment]
> @TallTed and all: we do have a [process](https://github.com/w3c/WebID/blob/main/CONTRIBUTING.md#processing-prs) in place that guarantees review times of up to 4 weeks for breaking changes and I remain steadfast in my quest for unanimity first and, if that cannot be achieved, as large a consensus as we can muster. While conversation can happen in bursts and evolve quickly, this doesn't automatically extend to how changes are made.
> 
> There's really no risk of this happening as long as we stick to the process we have in place. While I do share a certain frustration with how parts of the conversation are being brought forward, at the present moment there's no risk of anything being decided against general consensus.

In response to your message, I'd like to express my concerns with utmost respect but with necessary candor. The current approach to implementing substantial changes in a well-established, decade-old project raises significant apprehensions. Introducing major modifications without a more inclusive timeline for feedback disregards the extensive ecosystems and toolings that have matured over the years. Such a brief window for feedback, especially when it risks overlooking substantial objections, seems incongruous with the collaborative spirit and technical stability we've cherished.

Moreover, the decision to not revise the version number amidst these changes is perplexing. Maintaining the version as 1.0, despite fundamental alterations, not only confuses but also undermines the integrity of the project's evolution.

Crucially, the prioritization of this fork over the foundational task of defining WebID in a serialization-neutral manner contradicts the collective wisdom of the group. This leapfrogs the essential step of establishing a common understanding of WebID, crucial for ensuring compatibility and coherence across the board.

Additionally, the timing of these changes is particularly troubling. Forcing this transition in the narrow timeframe before the inception of the Solid WG deprives a significant stakeholder of their rightful input, potentially alienating a key segment of our community.

While change is inevitable and often necessary, the manner and sequence in which it is executed bear profound implications. The convergence of these concerns – the haste of implementation, disregard for versioning semantics, inversion of priority over serialization neutrality, and the ill-timed scheduling – collectively signify a departure from the principles of inclusivity, foresight, and respect for the community's legacy and future.

It is with a sense of responsibility towards the project's long-term health and community cohesion that I voice these objections. In the spirit of constructive dialogue and shared commitment to the project's success, I urge a reconsideration of the approach to ensure that it aligns with our collective aspirations and principles.

**Comment by jacoscaz** [comment]
_/chair hat on_

@melvincarvalho I don't like to ignore people but I can't keep addressing the same points ad infinitum. I will do so once again, here, and then stop answering to any further mention of these specific issues from you.

> the haste of implementation

So far, the only actual changes I've merged in since I became chair are: 

1. Formalizing our process and making it explicit.
2. Conversion to BikeShed.
3. Intended audience.

Other than this, the only thing I've done is trying to foster and locate consensus (see #37). I can't see how any of this classifies as haste (or implementation!).

If you believe that 4 weeks are not enough for major PRs to be adequately discussed, why don't you adopt a more proactive stance and propose a change to our process?

> disregard for versioning semantics

I've already made my point in https://github.com/w3c/WebID/issues/58#issuecomment-1925809369 . The 2014 ED states very clearly its own nature as an ED. You disagree, and I respect that, but I'm not going to change my mind just because you keep making these accusations.

> inversion of priority over serialization neutrality

At some point you will have to accept that not everyone believes in serialization neutrality. I can't do much more than quote a couple of recent posts to make you understand that, as chair, I can't just go with what you (or I, for that matter) want.

- https://github.com/w3c/WebID/issues/3#issuecomment-1879750583
- https://github.com/w3c/WebID/issues/17#issuecomment-1877196126

> Forcing this transition in the narrow timeframe before the inception of the Solid WG deprives a significant stakeholder of their rightful input

Solid people are welcome to join in the conversation here. Nobody is blocking them from doing so. In fact, I've been actively trying to reach out to them but, if I had to guess, they do not believe participation in this group to be a productive endeavor - and that is on all of us. I would like to prove them wrong and have them join the conversation once again but I can't do that on my own.

If you find any of my answers to be unacceptable, I beg you to please make the case for my resignation rather than continuing this manner of interaction. I understand disagreement and I would not think any less of you if you were to do so. I've said this many times: I'm happy to step down if the group would rather have someone else.

**Comment by melvincarvalho** [comment]
I am asking you to prioritize things in the right order.

1. define and specify webid, then select serialzations -- success
2. select serializations before webid is defined -- split group, and both halves fail

Its simply an ordering problem.  

Build the house then build the bathroom.  Dont build the bathroom then build the house.

The failure here is to understand that serializations can be selected after WebID is defined.

Just as SPARQL can be defined then used with any number of serializations.  But SPARQL cant be used with any serializations, before it is specified.

I need you to acknowledge the logic of the two step approach being consistent with everyone's needs.  Arguing that Turtle=MUST PRECLUDES defining defining WebID just is not a serious argument.  Defining and specifying WebID ENABLES the serialization conversation.

We just have to avoid putting the cart before the horse.  Specify WebID and everything else follows.

**Comment by melvincarvalho** [comment]
> What is the

@woutermont I'll not you've not yet provided a URL for the implementation that you outlined.  Does it still exist?

**Comment by melvincarvalho** [comment]
Let me add the mail list comments from Ruben Taelman:

https://lists.w3.org/Archives/Public/public-webid/2024Feb/0047.html

> I just want to briefly chime in and say that the “MUST on Turtle and JSON-LD on publishers”
 may represent consensus among the loudest and most present voices in this CG, 
**but it may not necessarily represent what less vocal people may be looking for**.

> But I strongly agree with Martynas’ comment on that WebID should be orthogonal to what RDF serialization is used.
WebID and RDF serialzation apply to different locations when observed in a layered structure.
It’s like expecting HTTP to **suddenly start requiring at least WiFi 802**.11ac to be used, these are orthogonal things.

emphasis mine.

**Comment by melvincarvalho** [comment]
This issue can now be closed, having proved sufficiently, that mandating conneg (e.g. via JSON-LD AND Turtle = MUST), is a breaking-change.

**Comment by csarven** [comment]
>Solid people are welcome to join in the conversation here. Nobody is blocking them from doing so. In fact, I've been actively trying to reach out to them but, if I had to guess, they do not believe participation in this group to be a productive endeavor - and that is on all of us. I would like to prove them wrong and have them join the conversation once again but I can't do that on my own.

You're severely mistaken.

All, if not, vast majority (perhaps except you) of the active/substantial contributors here *are* literally involved in Solid and in general have been driving the RWW space forward.

I for one helped to get the work continue in this repository so that the stalled work in the WebID CG can continue. You can read a brief background at https://github.com/w3c/WebID/issues/5#issue-1153260476 . I've also put in effort to work to continue even further in a W3C Recommendation track through the Solid WG charter proposal ( see proposal: https://github.com/solid/solid-wg-charter/issues/39#issue-1741064792 https://github.com/solid/solid-wg-charter/issues/39#issuecomment-1580886272 ) and efforts to unify communities. I've created/connected various issues/PRs, did all the open "grunt" editorial work (when no one even approached it for years.)

And I can refer to numerous other contributions from "Solid people" if that doesn't suffice. Your assertions are not only significantly mistaken, but I find it impolite. It is not you that needs to welcome "Solid people". "Solid people" have been active participants in this community long before your arrival, and if anyhting, it is they who welcomed you.

Start by speaking for yourself ( see [WebID CG communication guidelines](https://github.com/w3c/WebID/blob/main/CONTRIBUTING.md#communication) borrowed from [Solid CG's contribution guidelines](https://github.com/w3c-cg/solid/blob/main/CONTRIBUTING.md#communication)) instead of chalking up a whole set of participants as to what they think or don't about the current state of affairs.

Consider reading your statements again after perhaps considering the possibility that you may be in fact ignoring or downplaying recommendations that are spelled out from the start; clearly demonstrating "NIH"; anything and everything from chairing to editing..

**Comment by melvincarvalho** [comment]
> However, I would much rather suggest slowing down or pausing, rather than anything more drastic. In order to allow the soon-to-form Solid WG to shape, and they can inform us with vital new information.

Wishing to address this.  We have new information, the Solid WG is delayed and changing to the PUMPKIN WG.  That means that the original consideration of waiting for clarity on that.  We now have that clarity.  We need not be too heavily held up by PUMPKIN WG as it is unclear that they are near forming a WG, let alone taking WebID to REC.

> @melvincarvalho to clarify, would changing the title of the working draft https://w3c.github.io/WebID/spec/identity/index.html to _Web Identity and Discovery **1.1**_ cause you to withdraw your strong objection to conneg?

For the record, yes I think I could live with that, at this point.  Removing blockers to new work.

**Comment by melvincarvalho** [comment]
> And I can refer to numerous other contributions from "Solid people" if that doesn't suffice. Your assertions are not only significantly mistaken, but I find it impolite. It is not you that needs to welcome "Solid people". "Solid people" have been active participants in this community long before your arrival, and if anyhting, it is they who welcomed you.

@csarven please be respectful, and operate in good faith.  The tone here is not the best.  @jacoscaz has worked very hard to be inclusive.

> Consider reading your statements again after perhaps considering the possibility that you may be in fact ignoring or downplaying recommendations that are spelled out from the start; clearly demonstrating "NIH"; anything and everything from chairing to editing..

This is off-topic, and inappropriate.

> You're severely mistaken.
> 
> All, if not, vast majority (perhaps except you) of the active/substantial contributors here _are_ literally involved in Solid and in general have been driving the RWW space forward.

Again tone could be better.  Obviously there's quite an overlap with RWW, WebID and Solid (now PUMPKIN).  IMHO your ambition to be chair of the Solid WG was one more issue that it needed to handle.  You could achieve everything you wanted as an IE rather than a chair.  There wasnt a need to push for objections on the Solid WG based on chairs, when there were two perfectly good ones.  Let's put the animosity to one side and work together to make a better web.
